src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -239,7 +247,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       commitUnmount(node);
       // Visit children because they may contain more composite or host nodes.
       // Skip portals because commitUnmount() currently visits them recursively.
-      if (node.child !== null && node.tag !== HostPortal) {
+      if (
+        node.child !== null &&
+        // Drill down into portals only if we use mutation since that branch is recursive
+        (!config.mutation || node.tag !== HostPortal)
```

Comment:
The comment says we drill down *with* mutation but it seems like we only drill down *without* it?